tests/test-batch.php

```diff
@@ -18,79 +20,74 @@ public function tearDown() {
 
 	/**
 	 * Test name is included.
-	 *
-	 * @expectedExceptionMessage Batch name must be provided.
 	 */
 	public function test_register_batch_includes_name() {
-		$this->setExpectedException( 'Exception' );
+		try {
+			$batch_process = new Posts();
+			$batch_process->register( array(
+				'slug'     => 'test-anot22h12er-batch',
+				'type'     => 'post',
+				'callback' => 'my_callback_function',
+				'args'     => array(
+					'posts_per_page' => 10,
+					'post_type'      => 'post',
+				),
+			) );
+		} catch ( Exception $e ) {
+			$this->assertEquals( 'Batch name must be defined.', $e->getMessage() );
+		}
 
-		$batch_process = new Posts();
-		$batch_process->register( array(
-			'slug'     => 'test-anot22h12er-batch',
-			'type'     => 'post',
-			'callback' => 'my_callback_function',
-			'args'     => array(
-				'posts_per_page' => 10,
-				'post_type'      => 'post',
-			),
-		) );
 	}
 
 	/**
-	 * Test name is included.
-	 *
-	 * @expectedExceptionMessage Type of batch must be defined.
+	 * Test type is included.
 	 */
 	public function test_register_batch_includes_type() {
-		$this->setExpectedException( 'Exception' );
-
-		$batch_process = new Posts();
-		$batch_process->register( array(
-			'name'     => 'Hey',
-			'slug'     => 'test-anot22h12er-batch',
-			'callback' => 'my_callback_function',
-			'args'     => array(
-				'posts_per_page' => 10,
-				'post_type'      => 'post',
-			),
-		) );
+		try {
+			$batch_process = new Posts();
+			$batch_process->register( array(
+				'name'     => 'Hey',
+				'slug'     => 'test-anot22h12er-batch',
+				'callback' => 'my_callback_function',
+				'args'     => array(
+					'posts_per_page' => 10,
+					'post_type'      => 'post',
+				),
+			) );
+		} catch ( Exception $e ) {
+			$this->assertEquals( 'Batch type must be defined.', $e->getMessage() );
+		}
 	}
 
 	/**
-	 * Test array args is included.
-	 *
-	 * @expectedExceptionMessage An array of args must be defined.
+	 * Test callback defined
 	 */
 	public function test_register_batch_includes_callback() {
-		$this->setExpectedException( 'Exception' );
-
-		$batch_process = new Posts();
-		$batch_process->register( array(
-			'name'     => 'Hey',
-			'type'     => 'post',
-			'slug'     => 'test-anot22h12er-batch',
-			'args'     => array(
-				'posts_per_page' => 10,
-				'post_type'      => 'post',
-			),
-		) );
+		try {
+			$batch_process = new Posts();
+			$batch_process->register( array(
+				'name'     => 'Hey',
+				'type'     => 'post',
+				'slug'     => 'test-anot22h12er-batch',
+				'args'     => 0,
+			) );
+		} catch ( Exception $e ) {
+			$this->assertEquals( 'A callback must be defined.', $e->getMessage() );
+		}
 	}
 
 	/**
-	 * Test callback defined
-	 *
-	 * @expectedExceptionMessage A callback must be defined.
+	 * Test without arguments
 	 */
-	public function test_register_batch_includes_args() {
-		$this->setExpectedException( 'Exception' );
-
+	public function test_register_batch_no_args() {
```

Comment:
I think a better test name might be `test_register_batch_doesnt_require_args()` or `test_register_batch_works_without_args()` 🌴